code/src/App.js

```diff
@@ -1,9 +1,113 @@
-import React from 'react'
+import React, {useState} from 'react'
+
+
+import FavoriteBook from './Components/FavoriteBook';
+import HowManyBooks from './Components/HowManyBooks';
+import FavoriteGenre from './Components/FavoriteGenre';
+import Header from './Components/Header';
+import Summary from './Components/Summary';
+
+import './index.css'
+
+
 
 export const App = () => {
+
+  //function for submit button
+  const handleSubmit = (event) => {
+
+    event.preventDefault()
+    if (isSurveyComplete())
+    {
+      setDisplaySummary(true)
+    }
+    else{
+      //if survey isn't complete, it will alert and scroll to top.
+      alert("Survey not complete");
+      window.scroll(0,0);
+    }
+  }
+
+  const [number, setNumber] = useState("");
+  const [favoriteGenre, setFavoriteGenre] = useState("")
+  const [book, setBook] = useState("");
+  const [displaySummary, setDisplaySummary] = useState(false);
+
+  const onNumberChange = (event) => {
+    setNumber(event.target.value)
+  }
+
+  const onGenreChange = (event) => {
+    setFavoriteGenre(event.target.value)
+  }
+
+  const onBookChange = (event) => {
+    setBook(event.target.value);
+  }
```

Comment:
I might be wrong about this but I think these functions could be put inside each(??) (respektive!) components sending the set functions as props.